### PR TITLE
Bucket Logging: Accept Log Prefix as per S3 bucket logging

### PR DESCRIPTION
**Pull request by aspandey**
We need to change the log prefix restriction and accept it in accordance with AWS S3 guidelines.

As we do that, it is better to fetch the log prefix from the configuration of the source bucket instead of depending on log object names.

It's because the log prefix can have any character in it, and it would be difficult to have a delimiter that can segregate the log prefix from other strings in the log object name.

### Testing Instructions:
Testing Commands - 

ON end point run following command to create buckets and set bucket logging.

nb bucket create data.bucket
nb bucket create log.bucket
nb bucket create data.bucket-1
nb bucket create log.bucket-1

nb api bucket_api put_bucket_logging '{
  "name" : "data.bucket",
  "log_bucket": "log.bucket",
  "log_prefix": "data-bucket-logs"
}'

nb api bucket_api put_bucket_logging '{
  "name" : "data.bucket-1",
  "log_bucket": "log.bucket-1",
  "log_prefix": "data-b^*&^^@$%#&#)((&F/"
}'

nb api bucket_api get_bucket_logging '{
  "name" : "data.bucket-1"
}'

**On log-processor-container- cd /log/noobaa_bucket_logs/**
touch data.bucket_log.bucket_1576521528.log
touch data.bucket-1_log.bucket-1_1576524645645.log

After some time you should see these files uploaded to respective log buckets.

### CODE REVIEW

File: src/server/bg_services/bucket_logs_upload.js [hunk 1]
```diff
@@ -65,6 +65,24 @@ class BucketLogUploader {
...
+     */
+    _read_bucket_logging_prefix(bucket_name) {
+        const bucket = system_store.data.buckets.find(bucket => bucket.name.unwrap() === bucket_name);
+        if (!bucket) {
+            dbg.error('Could not find source bucket in system_store: ', bucket_name);
+            return "";
+        }
+        if (!bucket.logging) {
+            dbg.error('Found source bucket but bucket logging was not configured: ', bucket_name);
+            return "";
```
Comment (jackyalbo) [inline src/server/bg_services/bucket_logs_upload.js:81]: this is a legitimate prefix - so we shouldn't use this to mark that we got an error -  We should maybe handle the possible errors in the main function and skip for all of this possible issue.

Comment (aspandey) [reply src/server/bg_services/bucket_logs_upload.js:81]: Yes,  will change it.

Comment (aspandey) [reply src/server/bg_services/bucket_logs_upload.js]: done

File: src/server/bg_services/bucket_logs_upload.js [hunk 2]
```diff
@@ -65,6 +65,24 @@ class BucketLogUploader {
...
     }
 
+    /**
+     * Fetch the log prefix for the source bucket
+     * @param string bucket_name Name os the source bucket.
+     * @return string Configured log prefix for source bucket.
+     */
+    _read_bucket_logging_prefix(bucket_name) {
+        const bucket = system_store.data.buckets.find(bucket => bucket.name.unwrap() === bucket_name);
+        if (!bucket) {
```
[the remark this answers is not in the document]
Comment (aspandey) [reply src/server/bg_services/bucket_logs_upload.js]: > I don't think we should just skip in both these cases: the bucket can get deleted in the time between the log was created and uploaded so we should avoid this upload... and if no logging - also possible that the user disabled the bucket logging on this bucket - and also we should get rid of this logs and not to try and upload them.

done

[the remark this answers is not in the document]
Comment (jackyalbo) [reply src/server/bg_services/bucket_logs_upload.js]: > So, if a log_bucket is deleted then that case is being handled . A putObject call will fail, code will throw an exception and finally log object will be deleted anyway.

As we can check if the bucket exists ourselves, let's check in advance and avoid a lengthy S3 call when we can avoid it.

[the remark this answers is not in the document]
Comment (aspandey) [reply src/server/bg_services/bucket_logs_upload.js]: Yes Jacky. I have done like this in my latest patch.

**Review by jackyalbo — APPROVED** [review]

**Pull request merged**